File: CarLine.DataCleanUp/Controllers/CleanupController.cs
using CarLine.DataCleanUp.Services;
using Microsoft.AspNetCore.Mvc;

namespace CarLine.DataCleanUp.Controllers;

[ApiController]
[Route("api/[controller]")]
public class CleanupController : ControllerBase
{
    private readonly ILogger<CleanupController> _logger;
    private readonly DataCleanupService _cleanupService;

    public CleanupController(ILogger<CleanupController> logger, DataCleanupService cleanupService)
    {
        _logger = logger;
        _cleanupService = cleanupService;
    }

    [HttpPost("run")]
    public async Task<IActionResult> RunCleanup(CancellationToken cancellationToken)
    {
        _logger.LogInformation("Manual cleanup endpoint called.");
        var result = await _cleanupService.RunCleanupAsync(cancellationToken).ConfigureAwait(false);
        if (result.Success)
            return Ok(result);
        return StatusCode(500, result);
    }
}

File: CarLine.DataCleanUp.FSharp/CleanupController.fs
namespace CarLine.DataCleanUp.FSharp

open System.Threading
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging

[<ApiController>]
[<Route("api/[controller]")>]
type CleanupController(logger: ILogger<CleanupController>, cleanupService: DataCleanupService) =
    inherit ControllerBase()

    [<HttpPost("run")>]
    member this.RunCleanup(ct: CancellationToken) = task {
        logger.LogInformation("Manual cleanup triggered via API.")
        let! result = cleanupService.RunCleanupAsync(ct)
        return
            if result.Success then this.Ok(result)          :> IActionResult
            else               this.StatusCode(500, result) :> IActionResult
    }
